src/realm/mixed.cpp

```diff
@@ -372,27 +372,66 @@ int Mixed::compare_signed(const Mixed& b) const
     return compare(b);
 }
 
-template <class T>
-T Mixed::export_to_type() const noexcept
+template <>
+int64_t Mixed::export_to_type() const noexcept
+{
+    REALM_ASSERT(get_type() == type_Int);
```

Comment:
Maybe I'm not understanding this, but why is this not handling the other numeric types?

Reply:
If common type is Int then both values are integers.